### PR TITLE
Allow for async PubSub message handlers

### PR DESCRIPTION
**Pull request by mikeyk**

### CODE REVIEW

**Comment by mikeyk** [comment]
Ah looks like it's the Python 3.8 lint failing with the Python 3.10 union syntax; will fix

**Comment by alisaifee** [comment]
To get through some of the linting / formatting, you can use `make lint-fix` locally, assuming you've installed all the dependencies in `requirements/dev.txt`

**Comment by mikeyk** [comment]
thanks @alisaifee , looks clean now locally

File: coredis/commands/pubsub.py [hunk 1]
```diff
@@ -380,7 +379,10 @@ def handle_message(
                 handler = self.channels.get(message["channel"], None)
 
             if handler:
-                handler(message)
+                if asyncio.iscoroutinefunction(handler):
+                    await handler(message)
```
[the remark this answers is not in the document]
Comment (alisaifee) [reply coredis/commands/pubsub.py:383]: Actually tests & the change looks fine, I'll fix up the linting post merge.

**Comment by alisaifee** [comment]
I've renamed `PubSubHandler` to [SubscriptionCallback](https://coredis.readthedocs.io/en/latest/api.html#coredis.commands.pubsub.SubscriptionCallback). Just waiting for CI to do it's thing and will do a `4.10.0` release.

**Pull request merged**